### ISSUE TITLE
Mysql Flexible server administrator_login argument is not supporting hyphen character

### ISSUE THREAD

**Issue by r4kh1m** [hashicorp/terraform-provider-azurerm#26658]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Terraform Version

1.6.6

### AzureRM Provider Version

3.94.0

### Affected Resource(s)/Data Source(s)

azurerm_mysql_flexible_server

### Terraform Configuration Files

```hcl
resource "azurerm_mysql_flexible_server" "example" {
  name                   = "example"
  resource_group_name    = "sample-rg"
  location               = "westeurope"
  administrator_login    = "with-hyphen"
  administrator_password = "H@Sh1CoR3!"
}
```

### Debug Output/Panic Output

```shell
╷
│ Error: "administrator_login" must only contains characters, numbers or '_', got with-hyphen
│ 
│   with azurerm_mysql_flexible_server.example,
│   on main.tf line 32, in resource "azurerm_mysql_flexible_server" "example":
│   32:   administrator_login    = "with-hyphen"
```

### Expected Behaviour

`administrator_login` attribute accepts hyphen (-) character.

### Actual Behaviour

After [in-place auto-migration ](https://learn.microsoft.com/en-us/azure/mysql/migrate/migrate-single-flexible-in-place-auto-migration) from single to flexible server, the `Server admin login name` moved to the flexible one containing hyphen (-). 
It is impossible now to import the current Flexible server to the terraform because of the issue.

### Steps to Reproduce

`terraform plan`

### References
Blocked by - https://github.com/Azure/azure-mysql/issues/287
Similar issue:
#15128
#15280

**Comment by derekrprice** [comment]
This works as a workaround:

``` terraform
    lifecycle {
        # The API rejects the pre-existing hyphen in our username and changing the administrator_login
        # recreates the resource, destroying its data.  Just ignore this difference.
        ignore_changes = [administrator_login]
    }
```

**Comment by katbyte** [comment]
Thank you for taking the time to report this issue, and apologies that it has sat without a resolution for so long.

The error reported here (`must only contains characters, numbers or`, which was produced by the provider source as of v3.94.0) no longer exists in the current provider source — the relevant code has since been rewritten, so this report is no longer actionable as written.

As part of a cleanup of historical issues we are closing reports whose failing code paths have been rewritten. If you still run into this behaviour on the current major version (v5.x), please open a new issue with up-to-date reproduction details (feel free to reference this one) so it can be triaged with current context.

Thank you again for the report, and for your patience.
